### PR TITLE
Added option dns search domain

### PR DESCRIPTION
**Pull request by Yuav**

### CODE REVIEW

**Comment by alexjfisher** [comment]
@Yuav - Thanks!
Are you able to add a spec test for this?

**Comment by Yuav** [comment]
Fixed

File: templates/dhcpd.conf-header.erb [hunk 1]
```diff
@@ -15,6 +15,9 @@ log-facility <%= @logfacility %>;
 # ----------
 option domain-name "<%= @dnsdomain_real.first %>";
 option domain-name-servers <%= @nameservers.join(', ') %>;
+<% if @dnssearchdomains -%>
```
Comment (alexjfisher) [inline templates/dhcpd.conf-header.erb]: an empty array is also true...

Comment (Yuav) [inline templates/dhcpd.conf-header.erb]: Oh right, future parser. Nice catch! 

I'll add another test :-)

File: spec/classes/dhcp_spec.rb [hunk 2]
```diff
@@ -67,6 +67,19 @@
         is_expected.to contain_concat__fragment('dhcp-conf-header')
       end
 
+      context 'dns-search param' do
```
Comment (alexjfisher) [inline spec/classes/dhcp_spec.rb]: This should probably match the parameter name exactly

Comment (Yuav) [inline spec/classes/dhcp_spec.rb]: IDD

**Comment by Yuav** [comment]
Should be less bad now

File: templates/dhcpd.conf-header.erb [hunk 3]
```diff
@@ -15,6 +15,9 @@ log-facility <%= @logfacility %>;
 # ----------
 option domain-name "<%= @dnsdomain_real.first %>";
 option domain-name-servers <%= @nameservers.join(', ') %>;
+<% if @dnssearchdomains && @dnssearchdomains != [] -%>
```
Comment (alexjfisher) [inline templates/dhcpd.conf-header.erb]: `if @dnssearchdomains && !@dnssearchdomains.empty?` is possibly better from a style perspective?

Also matches other templates.  See https://github.com/voxpupuli/puppet-dhcp/blob/master/templates/dhcpd.conf.ntp.erb#L2

**Comment by Yuav** [comment]
Noted, fixed

**Comment by alexjfisher** [comment]
@Yuav Great! Looking good to me.  Looks like you need a rebase though.  Could you also squash your commits?  Thanks.

**Comment by Yuav** [comment]
Rebased.

**Comment by alexjfisher** [comment]
@Yuav Thanks, but it's still showing conflicts.  Perhaps you only rebased against your local master??  Did you forget to fetch and merge upstream master into your master first?

**Comment by Yuav** [comment]
History was messed up. Should be less broken now

File: spec/classes/dhcp_spec.rb [hunk 4]
```diff
@@ -71,6 +71,43 @@
...
+
+      context 'dnssearchdomains param empty array' do
+        let :params do
+          default_params.merge(
+            interfaces: ['eth0'],
+            dnssearchdomains: []
+          )
+        end
+
+        it 'has domain-search option with empty array' do
```
Comment (Yuav) [inline spec/classes/dhcp_spec.rb]: Should be fixed in PR #115

**Pull request merged**